src/components/menu/menu.ts

```diff
@@ -22,10 +24,21 @@ import {
 })
 export class MdMenu {
   private _showClickCatcher: boolean = false;
+  private _classList: Object;
+  positionX: 'before' | 'after' = 'after';
+  positionY: 'above' | 'below' = 'below';
 
   @Output() close = new EventEmitter;
   @ViewChild(TemplateRef) templateRef: TemplateRef<any>;
 
+  constructor(@Attribute('x-position') posX: 'before' | 'after',
+              @Attribute('y-position') posY: 'above' | 'below',
+              @Attribute('class') classes: string) {
```

Comment:
Yeah, was deliberately trying to keep it static in case a class affects the dimensions of the menu, which would throw off the position calculation if it changes later.  But if we are switching creation of the overlay to user interaction (rather than onInit), it doesn't matter. I'll update!